proxy/src/control/destination/background.rs

```diff
@@ -340,16 +351,27 @@ where
     fn reset_dns_query(
         &mut self,
         dns_resolver: &dns::Resolver,
-        delay: Duration,
+        deadline: Instant,
         authority: &DnsNameAndPort,
     ) {
+        // clamp the deadline down to the minimum TTL.
+        let deadline = if let Some(min_ttl) = self.dns_min_ttl {
+            let min_deadline = Instant::now() + min_ttl;
+            if deadline < min_deadline {
+                min_deadline
+            } else {
+                deadline
+            }
+        } else {
+            deadline
+        };
```

Comment:
I'm pretty sure that this "minimum of `Option<Duration>`" logic exists elsewhere; it would be great if we could abstract it into a reusable function.